### PR TITLE
WorkflowExecutor output processing

### PR DESCRIPTION
**Pull request by w-bonelli**
Addresses #230.

When processing outputs, only invoke Convert.ChangeType if the source type implements IConvertible.

### CODE REVIEW

File: src/WorkflowCore/Services/WorkflowExecutor.cs [hunk 1]
```diff
@@ -201,15 +201,12 @@ private void ProcessOutputs(WorkflowInstance workflow, WorkflowStep step, IStepB
...
                 var targetType = setter.Parameters.Last().Type;
-
                 var convertedValue = resolvedValue;
-                // We need to make sure the resolvedValue is of the correct type.
-                // However if the targetType is object we don't need to do anything and in some cases Convert.ChangeType will throw.
-                if (targetType != typeof(object))
+
+                if (typeof(IConvertible).IsAssignableFrom(resolvedValue.GetType()))
                 {
                     convertedValue = Convert.ChangeType(resolvedValue, targetType);
```
[the remark this answers is not in the document]
Comment (w-bonelli) [reply src/WorkflowCore/Services/WorkflowExecutor.cs]: Yes you're right, thank you; I've checked in a change that accommodates nullable types.

File: src/WorkflowCore/Services/WorkflowExecutor.cs [hunk 2]
```diff
@@ -201,15 +201,12 @@ private void ProcessOutputs(WorkflowInstance workflow, WorkflowStep step, IStepB
                 var data = workflow.Data;
                 var setter = ExpressionHelpers.CreateSetter(output.Target);
                 var targetType = setter.Parameters.Last().Type;
-
                 var convertedValue = resolvedValue;
-                // We need to make sure the resolvedValue is of the correct type.
-                // However if the targetType is object we don't need to do anything and in some cases Convert.ChangeType will throw.
-                if (targetType != typeof(object))
+
+                if (typeof(IConvertible).IsAssignableFrom(resolvedValue.GetType()))
```
[the remark this answers is not in the document]
Comment (w-bonelli) [reply src/WorkflowCore/Services/WorkflowExecutor.cs]: Trying to convert `string` to `Enum` would complicate the logic involved here; for now I've just used `Expression.Convert`, which appears to handle both the the `IConvertible` and the `Nullable` cases.

**Comment by w-bonelli** [comment]
Reimplemented via `Expression.Convert` instead of `Convert.ChangeType`

**Comment by danielgerlag** [comment]
#245

**Pull request closed without merging**